metaflow/metaflow_config.py

```diff
@@ -69,6 +69,12 @@
 S3_ENDPOINT_URL = from_conf("S3_ENDPOINT_URL")
 S3_VERIFY_CERTIFICATE = from_conf("S3_VERIFY_CERTIFICATE")
 
+# Dictionary containing configurable S3 upload settings
+# e.g. ServerSideEncryption, Tagging, StorageClass, etc. 
+# Use a JSON string to represent this parameter within an environment variable
+# and use a dictionary to represent this parameter within a JSON config file.
+S3_UPLOAD_ARGS = from_conf("UPLOAD_ARGS_S3", {})
```

Comment:
What would this look like if we passed in S3_UPLOAD_ARGS through the cli?

Since we can do things like: `METAFLOW_SERVICE_URL=http://localhost:10001 METAFLOW_DEFAULT_METADATA="service" python test-aws.py run`

Reply:
Just tested that and it works fine as well :)